### PR TITLE
Create Pets

### PR DESCRIPTION
**Pull request by anhdtp-1773**
==> T
==> Redmine: https://edu-redmine.sun-asterisk.vn/issues/8997
==> Attachment File:
Screenshot from 2019-04-01 10-48-07
Screenshot from 2019-04-01 10-49-25
Screenshot from 2019-04-01 10-49-41
Screenshot from 2019-04-01 10-50-03
Screenshot from 2019-04-01 10-50-16
Screenshot from 2019-04-01 10-50-45
Screenshot from 2019-04-01 10-51-13
Screenshot from 2019-04-01 10-53-17
Screenshot from 2019-04-01 10-53-50

### CODE REVIEW

**Comment by anhdtp-1773** [comment]
ready

File: app/models/pet.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 class Pet < ApplicationRecord
-  belongs_to :user
   has_many :order_details
+
+  PET_PARAMS = [:name, :description].freeze
+
+  validates :name, presence: true, length: {maximum: Settings.max_name_lenght}
+  validates :description, presence: true,
+                      length: {maximum: Settings.max_sumary_length}
```
Comment (huuphuc93) [inline app/models/pet.rb]: indent 2 spaces

File: app/models/user.rb [hunk 2]
```diff
@@ -8,15 +8,23 @@ class User < ApplicationRecord
...
+  USER_PARAMS = [:name, :phone, :address, :email, :password, :password_confirmation].freeze
   PASSWORD_PARAMS = [:password, :password_confirmation]
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  NUMBER_REGEX = /\d[0-9]\)*\z/
 
   validates :name, presence: true, length: {maximum: Settings.max_name_lenght}
-  validates :email, presence: true, length: {maximum: Settings.max_email_lenght},
-   format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  validates :address, presence: true,
+                      length: {maximum: Settings.max_address_lenght}
```
Comment (huuphuc93) [inline app/models/user.rb]: indent 2 spaces

**Comment by anhdtp-1773** [comment]
ready

File: app/views/admin/posts/index.html.erb [hunk 3]
```diff
@@ -22,7 +22,10 @@
             <tr>
               <td><%= create_index params[:page], index, Settings.per_page %></td>
               <td><%= post.title %></td>
-              <td><%= post.content %></td>
+              <td>
+                <%= truncate post.content, length: 160%>
```
Comment (huuphuc93) [inline app/views/admin/posts/index.html.erb]:
```suggestion
                <%= truncate post.content, length: 160 %>
```
160 => Settings

File: app/views/admin/pets/_pet.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    <%= pet.sumary %>
+    <%= image_tag pet.picture.url if pet.picture? %>
+  </span>
+  <span class="timestamp">
+    <%= t ".pet" %>
+    <%= time_ago_in_words(pet.created_at) %>
+    <%= t ".ago" %>
+    <% if current_user.admin? %>
+      <%= link_to t(".delete"), pet, method: :delete,
+                                       data: { confirm: t(".are_u_sure") } %>
```
Comment (huuphuc93) [inline app/views/admin/pets/_pet.html.erb]: indent

**Comment by anhdtp-1773** [comment]
ready

**Pull request merged**